nautobot/core/testing/views.py

```diff
@@ -555,8 +574,37 @@ def test_delete_object_with_permission(self):
                 self.assertEqual(len(objectchanges), 1)
                 self.assertEqual(objectchanges[0].action, extras_choices.ObjectChangeActionChoices.ACTION_DELETE)
 
+            if hasattr(self.model, "notes"):
+                # Verify Notes deletion
+                with self.assertRaises(ObjectDoesNotExist):
+                    Note.objects.get(assigned_object_id=instance.pk)
+
+                note_objectchanges = ObjectChange.objects.filter(changed_object_id__in=instance_note_pk_list)
+                self.assertEqual(len(note_objectchanges), 3)
+                self.assertEqual(note_objectchanges[0].action, extras_choices.ObjectChangeActionChoices.ACTION_DELETE)
+
         @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
         def test_delete_object_with_permission_and_xwwwformurlencoded(self):
```

Comment:
Probably overkill to add to this test case since its only difference that I see from the previous one is that it adds a different request content-type?